### PR TITLE
fix for test suite

### PR DESCRIPTION
**Pull request by JonSonesen**
Hey,
So here is what I came up with for one command test all solution. It is not very elegant but it runs all the tests using 'python setup.py test'

### CODE REVIEW

**Comment by rodneykeeling** [comment]
Fixes #42

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+recursive-exclude discogs_client/tests *symlinks*
```
Comment (rodneykeeling) [inline MANIFEST.in]: Could you make this reference the exact file?
i.e., `exclude discogs_client/tests/make_symlinks.py`

**Comment by rodneykeeling** [comment]
Did some quick testing, and I don't think we actually need the `test_all.py` file. Looks like it works fine and passes all 27 tests with just the `MANIFEST.in` file added.

**Comment by JonathonSonesen** [comment]
awesome I will remove it then. Would you prefer that rebase and  edit the commit or just do a new one?

File: MANIFEST.in (hunk 1 above)
Comment (JonathonSonesen) [inline MANIFEST.in]: yeah no problem!

**Comment by rodneykeeling** [comment]
Go ahead and rebase into 1 commit. Major thanks for diving into this! :beers:

**Comment by JonathonSonesen** [comment]
Hey Rodney,
I did find that the test_all.py file is required to run the tests. If you delete it then run 'python setup.py sdist' unpack it and run 'python setup.py test' the tests will not run. 
Please take a look and let me know what you think.
Thanks

**Comment by rodneykeeling** [comment]
Hmm, did you change `test_suite='discogs_client.tests.test_all'` back to `test_suite='discogs_client.tests'`? The tests seem to run fine and pass with just the MANIFEST.in file addition.

**Comment by JonathonSonesen** [comment]
Yeah I did and I receive a message: 
Traceback (most recent call last):
<pre><code>  File "setup.py", line 29, in <module>
    'discogs_client',
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/lib/python2.7/dist-packages/setuptools/command/test.py", line 142, in run
    self.with_project_on_sys_path(self.run_tests)
  File "/usr/lib/python2.7/dist-packages/setuptools/command/test.py", line 122, in with_project_on_sys_path
    func()
  File "/usr/lib/python2.7/dist-packages/setuptools/command/test.py", line 163, in run_tests
    testRunner=self._resolve_as_ep(self.test_runner),
  File "/usr/lib/python2.7/unittest/main.py", line 94, in __init__
    self.parseArgs(argv)
  File "/usr/lib/python2.7/unittest/main.py", line 149, in parseArgs
    self.createTests()
  File "/usr/lib/python2.7/unittest/main.py", line 158, in createTests
    self.module)
  File "/usr/lib/python2.7/unittest/loader.py", line 130, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python2.7/unittest/loader.py", line 103, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/usr/lib/python2.7/dist-packages/setuptools/command/test.py", line 37, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/usr/lib/python2.7/unittest/loader.py", line 91, in loadTestsFromName
    module = __import__('.'.join(parts_copy))
  File "/home/doombot/forks/discogs_client/dist/discogs-client-2.2.0/discogs_client/tests/make_symlinks.py", line 8, in <module>
    root, next = name.split('?')
ValueError: need more than 1 value to unpack
</code></pre>

**Comment by rodneykeeling** [comment]
Ah, yeah, I **think** that might have to do with the ordering of the include/exclude directives. Try this:

```
recursive-include discogs_client/tests *.py
recursive-include discogs_client/tests/res *.json
exclude discogs_client/tests/make_symlinks.py
```

**Comment by JonathonSonesen** [comment]
ah! yeah that did it. Sorry for the confusion. I was clearly not thinking that one through very thoroughly :blush:

**Comment by cburmeister** [comment]
Hey Jon! :beers:

File: discogs_client/tests/test_all.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    def test_all(self):
+        from discogs_client.tests import test_core, test_models, test_fetchers, test_utils
+        suite = unittest.TestSuite(test_core.suite())
+        unittest.TextTestRunner().run(suite) 
+        suite = unittest.TestSuite(test_models.suite())
+        unittest.TextTestRunner().run(suite) 
+        suite = unittest.TestSuite(test_fetchers.suite())
+        unittest.TextTestRunner().run(suite) 
+        suite = unittest.TestSuite(test_utils.suite())
+        unittest.TextTestRunner().run(suite)
```
Comment (cburmeister) [inline discogs_client/tests/test_all.py]: You can put these suites into a list and loop through them:

```
for test in [test_core, test_models, test_fetchers, test_utils]:
    unittest.TextTestRunner().run(test) 
```

Comment (JonathonSonesen) [inline discogs_client/tests/test_all.py]: Hey,
Thanks for the tip, looks like it's completely unneeded at this point. Just needs the MANIFEST.in that Rodney talked about, just working on rebasing these into one commit, do you recommend squashing them into one commit or just deleting the ones that aren't the most recent valid commit?

**Comment by JonathonSonesen** [comment]
Dang! I just screwed that rebase up....sorry

**Comment by rodneykeeling** [comment]
Awesome, looks good to me! Thanks again for tackling this issue. :+1:

**Pull request merged**